### PR TITLE
[NativeAOT/macOS] Allow lldb to skip over null reference exceptions

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

**Review by janvorli — APPROVED** [review]
LGTM, thank you.

File: src/coreclr/nativeaot/Runtime/unix/HardwareExceptions.cpp [hunk 1]
```diff
@@ -587,6 +594,27 @@ bool InitializeHardwareExceptionHandling()
         return false;
     }
 
+#if defined(HOST_OSX)
+	// LLDB installs task-wide Mach exception handlers. XNU dispatches Mach
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/unix/HardwareExceptions.cpp:598]: Does this problem exists in regular CoreCLR as well - should the same fix be made there?

Comment (janvorli) [reply src/coreclr/nativeaot/Runtime/unix/HardwareExceptions.cpp:598]: It doesn't. We don't use signals for sigsegv / arithmetic errors there and we use a mach exception handler.

**Comment by jkotas** [comment]
@filipnavara Thank you!

**Pull request merged**